controllers/fenceagentsremediation_controller.go

```diff
@@ -68,69 +75,68 @@ type FenceAgentsRemediationReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/reconcile
 func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	r.Log.Info("started reconcile")
-	defer r.Log.Info("finished reconcile")
+	r.Log.Info("Begin FenceAgentsRemediation Reconcile")
+
+	// Fetch the FenceAgentsRemediationTemplate instance
+	emptyResult := ctrl.Result{}
+	key := client.ObjectKey{Namespace: req.Namespace, Name: fenceAgentsTemplateName}
+	farTemplate := &v1alpha1.FenceAgentsRemediationTemplate{}
+	if err := r.Client.Get(ctx, key, farTemplate); err != nil {
+		if apiErrors.IsNotFound(err) {
+			// FAR Template is not found, stop reconciling
+			r.Log.Info("FAR Template CR is not found - nothing to do", "CR Name", req.Name, "CR Namespace", req.Namespace)
+			return emptyResult, nil
+		}
+		r.Log.Error(err, "failed to get FAR Template CR")
+		return emptyResult, err
+	}
 
+	// Fetch the FenceAgentsRemediation instance
+	r.Log.Info("Fetch FAR CR")
 	far := &v1alpha1.FenceAgentsRemediation{}
-	if err := r.Get(ctx, req.NamespacedName, far); err != nil {
+	if err := r.Client.Get(ctx, req.NamespacedName, far); err != nil {
 		if apiErrors.IsNotFound(err) {
 			// FAR is deleted, stop reconciling
-			r.Log.Info("Fence Agents Remediation not found, nothing to do")
-			return ctrl.Result{}, nil
+			r.Log.Info("FAR CR is deleted - nothing to do", "CR Name", req.Name, "CR Namespace", req.Namespace)
```

Comment:
I don't think we can assume it's deleted :thinking: 
I think creation of the template alone will also get us here.

Reply:
Creation or deletion of FARTemplete won't trigger the reconcile since it listen/watch for FAR CRs.
If we have reached to [line 97](https://github.com/medik8s/fence-agents-remediation/pull/11/files/260d55261948ed3d0fee45279adee1d7f3a30640#diff-05147ee3f0d225b5a60754b8b9bff2c569deb0a9d7eaddae33af680291d493a1R97) then FAR CR has been created or deleted, and were able to fetch the FARTemplete. Otherwise we should have received an error from [line 84 ](https://github.com/medik8s/fence-agents-remediation/pull/11/files/260d55261948ed3d0fee45279adee1d7f3a30640#diff-05147ee3f0d225b5a60754b8b9bff2c569deb0a9d7eaddae33af680291d493a1R84) and never reach this far ;)

Reply:
what a nice example for why we shouldn't use the template CR for anything but being the literally template for the real remediation CRs. It only causes confusion...

Reply:
https://github.com/medik8s/fence-agents-remediation/pull/16 will hopefully resolve that 😸